### ISSUE TITLE
ABI version of plugin: 'calc.so' does not match: 00000006 expecting: 00000007

### ISSUE THREAD

**Issue by revsuine** [svenstaro/rofi-calc#116]
Hi there,

I updated `rofi-lbonn-wayland` to `1.7.5+wayland3-1`, which causes the following error to appear whenever you either run rofi with `calc` mode added to my `config.rasi`, or run with `rofi -show calc -modi calc -no-show-match -no-sort`:

```
(process:21205): Rofi-WARNING **: 11:59:57.309: ABI version of plugin: 'calc.so' does not match: 00000006 expecting: 00000007
```

This is the same issue as #107, however this only started happening for me after the latest `rofi-lbonn-wayland` update.

I have tried rebuilding both `rofi-lbonn-wayland` (again after the update) and `rofi-calc`, both to no avail.

**Comment by svenstaro** [comment]
I don't believe this has anything to do with this plugin's code. It's an ABI error and so a rebuild should fix it. Are you sure you rebuilt it correctly?

**Comment by svenstaro** [comment]
You also need to rebuild rofi-calc using makepkg against your rofi. If you install it from Arch repos, you will just get a version that is compiled against the official rofi.

**Comment by revsuine** [comment]
Thank you, that fixed it

**Comment by untainsYD** [comment]
> Thank you, that fixed it

so, how did u manage to fix it?

**Comment by revsuine** [comment]
@untainsYD clone https://gitlab.archlinux.org/archlinux/packaging/packages/rofi-calc, cd into it, `makepkg -si`

**Comment by untainsYD** [comment]
> @untainsYD clone https://gitlab.archlinux.org/archlinux/packaging/packages/rofi-calc, cd into it, `makepkg -si`
> 
> (if on Arch, if you're on another distro rebuild `rofi-calc` for that distro)

thanks

**Comment by Rolv-Apneseth** [comment]
Just gonna add here for anyone else stumbling upon this issue, I added [rofi-calc-git](https://aur.archlinux.org/packages/rofi-calc-git) to the AUR so you should just be able to install that as it will clone the repo and build it from source.

If anyone has experience with writing `PKGBUILD`, I'd appreciate any feedback, as it installs just fine for me but maybe I missed something.

**Comment by revsuine** [comment]
@Rolv-Apneseth jsyk, [the `extra` package](https://gitlab.archlinux.org/archlinux/packaging/packages/rofi-calc/-/blob/main/PKGBUILD) also builds from source already, however it does so for a tagged release rather than the latest git commit (which yours builds). e.g. I used the `extra/rofi-calc` package to reinstall, pacman just didn't do it for me (I assume as I already had the build files cached somewhere?), I had to clone the `PKGBUILD` and run `makepkg -si` for it to happen. (Not saying you should take the package down or anything, it still offers an option for people wanting to install the latest git commit rather than the latest tagged commit, but just that the existing makepkg also builds from source, just not the latest commit)

Btw, you should add `libqualculate` as a dependency. Otherwise, LGTM

**Comment by Rolv-Apneseth** [comment]
Ah I didn't know that, thanks. So I assume clearing cache / using `paru` with `--rebuild` should do it then.

And thanks I'll add that dependency

**Comment by the-pink-hacker** [comment]
I had this issue with NixPkgs since I'm using the wayland version. This seemed to fix it for me:
```nix
programs.rofi = {
    enable = true;
    package = pkgs.rofi-wayland;
    plugins = with pkgs; [
        (rofi-calc.override {
            rofi-unwrapped = pkgs.rofi-wayland-unwrapped;
        })
    ];
};
```
